packages/desktop-client/src/components/accounts/TransactionsTable.js

```diff
@@ -1787,6 +1843,77 @@ export let TransactionTable = forwardRef((props, ref) => {
     id => splitsExpanded.dispatch({ type: 'toggle-split', id }),
     [splitsExpanded.dispatch],
   );
+  let [field, setField] = useState('date');
+  let [ascDesc, setAscDesc] = useState('asc');
+
+  function onSortTable(headerClicked) {
+    let transform =
+      headerClicked.currentTarget.id === 'payment' ||
+      headerClicked.currentTarget.id === 'deposit'
+        ? 'amount'
+        : headerClicked.currentTarget.id;
+    if (transform === field) {
+      ascDesc === 'desc' ? setAscDesc('asc') : setAscDesc('desc');
+    } else {
+      setField(transform);
+      transform === 'amount' || transform === 'date'
+        ? setAscDesc('asc')
+        : setAscDesc('desc');
+    }
+  }
+
+  let category = props.categoryGroups.flatMap(cat => {
+    return cat.categories;
+  });
+  let fields = field + 's';
+
+  function sortStrings() {
+    let sortDesc = transactions
+      .map(e => {
+        return {
+          ...e,
+          fieldName: props[fields].find(f => f.id === e[field])?.name ?? '',
+        };
+      })
+      .sort((a, b) => a.fieldName > b.fieldName);
+
+    transactions = ascDesc === 'desc' ? sortDesc : sortDesc.reverse();
+  }
+
+  function sortCat() {
+    let sortDesc = transactions
+      .map(e => {
+        return {
+          ...e,
+          fieldName: category.find(f => f.id === e[field])?.name ?? '',
+        };
+      })
+      .sort((a, b) => a.fieldName > b.fieldName);
```

Comment:
Sorters are supposed to return a number (`-1` for less than, `0` for equal, `1` for greater than) rather than a boolean. This might be causing the instability, but I’m not sure. For strings, I would recommend using `a.localeCompare(b)`. For numbers I think the convention tends to be `a - b`.